### PR TITLE
[builddep] fix build Ubuntu 16.4 Xenial

### PR DESCRIPTION
**Pull request by AlwinEsch**
The following build dependencies changes fix build on the the next Ubuntu Version 16.4 (Xenial) which use GCC 5 and bring compile errors on libssh and libgpg-error.

Also added temporary  libgpg-error-1.20.tar.bz2 for upload to sources storage.

[edit] temporary file removed, due to correct link with related home page (thanks @fritsch )

https://www.gnupg.org/(fr)/download/index.html#libgpg-error

### CODE REVIEW

**Comment by AlwinEsch** [comment]
Here the related log files:
https://www.dropbox.com/s/1efr1g5pqfkddkm/libssh-compile-with-gcc5.log?dl=0
https://www.dropbox.com/s/u6e0ujgujg5qpa6/libgpg-error-compile-with-gcc5.log?dl=0

**Comment by AlwinEsch** [comment]
Can someone upload the https://www.gnupg.org/ftp/gcrypt/libgpg-error/libgpg-error-1.20.tar.bz2 to xbmc/build-deps/sources?

**Comment by wsnipex** [comment]
done. give it ~30 mins to propagate to mirrors

**Comment by AlwinEsch** [comment]
Thanks @wsnipex.

Jenkins build this please.

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins has failed everywhere with "No JDK named ‘null’ found", due to my changes no failure seen.

I try again (and without typo)

jenkins build this please again

**Comment by AlwinEsch** [comment]
Can someone look on Jenkins, for OSX and Android it fails but can't find the related error on logs.

**Comment by wsnipex** [comment]
droid x86:
>checking whether grantpt is declared without a macro... libtool: compile:  /home/jenkins/android-dev//android-toolchain-x86-17-r10d/bin/i686-linux-android-gcc -DHAVE_CONFIG_H -I. -I.. -DANDROID -Os -fexceptions -I/home/jenkins/workspace/Android-X86/tools/depends/xbmc-depends/i686-linux-android-android-17/include/android-17 -isystem /home/jenkins/workspace/Android-X86/tools/depends/xbmc-depends/i686-linux-android-android-17/include -DANDROID -Os -fexceptions -I/home/jenkins/workspace/Android-X86/tools/depends/xbmc-depends/i686-linux-android-android-17/include/android-17 -isystem /home/jenkins/workspace/Android-X86/tools/depends/xbmc-depends/i686-linux-android-android-17/include -MT zip_fopen_index.lo -MD -MP -MF .deps/zip_fopen_index.Tpo -c zip_fopen_index.c -o zip_fopen_index.o
./gpg-error.h.in:402: error including `./syscfg/lock-obj-pub.linux-android.h': No such file or directory

**Comment by mkortstiege** [comment]
@AlwinEsch you send the patches upstream yet?

**Comment by Jalle19** [comment]
For the record, Ubuntu 15.10 uses GCC 5 as well.

**Comment by AlwinEsch** [comment]
@mkortstiege what you mean with upstream? Things like this are a bit new for me. The one needed patch is only added on request for there "tools/depends/target/libssh/fix-gcc-5-compile.patch"

Thanks @Jalle19, has not tested on my Laptop with 15.10 only with 15.04 installed, thought only on my Mupel to use the newest and see everything works and how something like this can be tested.

**Comment by wsnipex** [comment]
he means send the patch to libssh, so it can be included in the next release:
https://red.libssh.org/projects/libssh/issues?set_filter=1&tracker_id=1

**Comment by AlwinEsch** [comment]
Ah, now I understand. Is this way OK? Has looked over the changes from Ubuntu here http://packages.ubuntu.com/xenial/libssh-dev and reworked the patch from them to use on our Version.

**Comment by mkortstiege** [comment]
Sounds ok to me. @wsnipex, objections?

**Comment by AlwinEsch** [comment]
Have added a script to generate for 'libgpg-error' the missing headers on OS X and so on.

The Makefile generate the needed binary and calls then the script.
It must be improved with a "```if```" to do it only there where needed.

Is this way OK. Have currently no other idea. This libgpg-error version is a bit different as the old before.

**Comment by AlwinEsch** [comment]
Has added for libssh this issue https://red.libssh.org/issues/206

Also was my Mupel now the last change OK on Mac OS X.

File: tools/depends/target/libgpg-error/generate-posix-lock-obj.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+PLATFORM=$1
+
+$PLATFORM/src/gen-posix-lock-obj > $PLATFORM/src/syscfg/tmp.h
+mv $PLATFORM/src/syscfg/tmp.h $PLATFORM/src/syscfg/$(cat $PLATFORM/src/syscfg/tmp.h | head -1 | awk -F '## ' '{print $2}')
```
Comment (wsnipex) [inline tools/depends/target/libgpg-error/generate-posix-lock-obj.sh]: you can do that directly in the Makefile, would be a bit more readable to have it all in one place.
cosmetics:
mv $PLATFORM/src/syscfg/tmp.h $PLATFORM/src/syscfg/$($(AWK) 'NR==1 { print $2 }' $PLATFORM/src/syscfg/tmp.h)

if you prefer your version, please use $(AWK) anyway and put it in the Makefile

**Comment by wsnipex** [comment]
jenkins build this please

File: tools/depends/target/libgpg-error/Makefile [hunk 2]
```diff
@@ -27,6 +27,18 @@ $(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
 	cd $(PLATFORM); $(CONFIGURE)
 
 $(LIBDYLIB): $(PLATFORM)
+ifeq ($(OS),osx)
+	$(MAKE) -C $(PLATFORM)/src gen-posix-lock-obj
+	$(PLATFORM)/src/gen-posix-lock-obj > $(PLATFORM)/src/syscfg/tmp.h
+	mv $(PLATFORM)/src/syscfg/tmp.h $(PLATFORM)/src/syscfg/$($(AWK) 'NR==1 {print $2}' $(PLATFORM)/src/syscfg/tmp.h)
```
Comment (wsnipex) [inline tools/depends/target/libgpg-error/Makefile]: I'm sorry, I though $(AWK) was defined in depends...
this is how it works:
>mv $(PLATFORM)/src/syscfg/tmp.h $(PLATFORM)/src/syscfg/$$(awk 'NR==1 {print $$2}' $(PLATFORM)/src/syscfg/tmp.h)

you really need the $$

Comment (AlwinEsch) [inline tools/depends/target/libgpg-error/Makefile]: No I know it, you are a bit late ;-D

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
OK still bad on has forgot the $$ on start "$($(AWK)", Mac has not understand $(AWK) and was empty.
@wsnipex sorry as not seen message

Has checked now before and build on Android x86 and Mac OS X was OK. Then the last time:
jenkins build this please again

**Comment by AlwinEsch** [comment]
But now!

The generation of header is not possible for Android (or I has not found a way). Is changed now to a copy of the related arm header to have general for all Androids. Has looked over VLC, they use the same way.

For Mac OS X is leaved on the way before which was OK.

jenkins build this please again, sorry!

**Comment by AlwinEsch** [comment]
Had forgotten a temporary code change to remove.

jenkins build this please again, sorry, sorry!

**Comment by AlwinEsch** [comment]
OK to merge in??

**Pull request merged**